central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen_impl.go

```diff
@@ -114,12 +115,17 @@ func (rg *complianceReportGeneratorImpl) ProcessReportRequest(req *ComplianceRep
 
 	log.Infof("Processing report request %s", req)
 
-	snapshot, found, err := rg.snapshotDS.GetSnapshot(req.Ctx, req.SnapshotID)
-	if err != nil {
-		return errors.Wrap(err, "unable to retrieve the snapshot from the store")
-	}
-	if !found {
-		return errors.New("unable to find snapshot in the store")
+	var snapshot *storage.ComplianceOperatorReportSnapshotV2
+	if features.ScanScheduleReportJobs.Enabled() {
+		var found bool
+		var err error
+		snapshot, found, err = rg.snapshotDS.GetSnapshot(req.Ctx, req.SnapshotID)
+		if err != nil {
+			return errors.Wrap(err, "unable to retrieve the snapshot from the store")
+		}
+		if !found {
+			return errors.New("unable to find snapshot in the store")
+		}
```

Comment:
If the feature flag is disabled, then the snapshot will be nil. I followed all the following code that may use the `snapshot` and I see that we should be safe as all those places also check for the feature-flag (although the IDE is not so clever and issues warnings that `snapshot` may be nil).

The issue with that approach is that we have many checks of this feature-flag that are spread through the code. This will work for now, but the testability and the future maintainability of this code gets harder. I am thinking if it maybe would be better to check the FF once and adapt the later code to have a different behavior if the snapshot is nil?

Reply:
Yeah that's much better. I've changed to handle the `nil` snapshot instead of checking for the FF.

Reply:
Thanks for improving!